From 008f89cffa31a2bab640f2be8db42fbc3f322833 Mon Sep 17 00:00:00 2001
From: Guilhem Saurel <guilhem.saurel@laas.fr>
Date: Wed, 7 Apr 2021 12:51:54 +0200
Subject: [PATCH] fix boost bind placeholders
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

otherwise, with a recent Boost:

/usr/include/boost/bind.hpp:36:1: note: « #pragma message: The practice
of declaring the Bind placeholders (_1, _2, ...) in the global namespace
is deprecated. Please use <boost/bind/bind.hpp> + using namespace
boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain
the current behavior. »
---
 tests/tools-obstacle.hh | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/tests/tools-obstacle.hh b/tests/tools-obstacle.hh
index 5e4ab6b..7c98d09 100644
--- tests/tools-obstacle.hh
+++ tests/tools-obstacle.hh
@@ -38,6 +38,13 @@
 #include <hpp/rbprm/planner/oriented-path-optimizer.hh>
 #include <hpp/rbprm/dynamic/dynamic-path-validation.hh>
 
+#if BOOST_VERSION / 100 % 1000 >= 60
+#include <boost/bind/bind.hpp>
+using namespace boost::placeholders;
+#else
+#include <boost/bind.hpp>
+#endif
+
 using namespace hpp;
 using namespace hpp::core;
 
